test/support/factory/prediction.ex

```diff
@@ -0,0 +1,12 @@
+defmodule Test.Support.Factory.Prediction do
+  @moduledoc """
+  Generated fake data for %RoutePattern{}
+  """
+  use ExMachina
+
+  alias Predictions.Prediction
+
+  def prediction_factory do
+    %Prediction{}
```

Comment:
In the above tests, you are building a prediction and passing in every single value. I'm fairly certain that not setting anything here will set them all to struct default which is probably nil. So, passing in values set to nil won't do anything. You should set reasonable defaults here and then you only have to override ones that are necessary for the tests. Makes the tests shorter and clearer.